prow/cmd/sinker/main.go

```diff
@@ -218,18 +221,26 @@ func (c *controller) clean() {
 		}
 		maxPodAge := c.config().Sinker.MaxPodAge
 		for _, pod := range pods.Items {
-			if _, ok := isFinished[pod.ObjectMeta.Name]; !ok {
-				// prowjob is not marked as completed yet
+			clean := !pod.Status.StartTime.IsZero() && time.Since(pod.Status.StartTime.Time) > maxPodAge
+			if !isFinished.Has(pod.ObjectMeta.Name) {
+				// prowjob exists and is not marked as completed yet
 				// deleting the pod now will result in plank creating a brand new pod
+				clean = false
+			}
+			if !isExist.Has(pod.ObjectMeta.Name) {
```

Comment:
isn't this a strict subset of `!isFinished.Has()` ?

Reply:
yes, and that's why we want to check it again after `!isFinished.Has(pod.ObjectMeta.Name)`

Reply:
Uh, I mean there is no use in checking it since it is a strict subset of the previous check?

Reply:
ehh hummm,

if I have all = {a, b, c, d}, finished = {a, b}, exist = {a, b, c} =>
^finished = {c, d}, ^exist = {d} =>
we want to not clean ^finished ∩ exist, so
d is in ^finished so we don't want to clean , but however
d is in ^exist so we do want to clean (the subset has a different condition here)
...
right?

Reply:
Finished must be a superset of exist

Reply:
Like we loop over what exists to check if each item of that is finished. no?

Reply:
More specifically -- `isFinished` is the set of `ProwJob` names that are `Complete()` whereas `isExist` is the full set of `ProwJob` names on the cluster.

Reply:
yes that's all correct - but we want to clean up anything in `!isExist` regardless of `!isFinished` right? (`!isExist` is a subset of `!isFinished` which wants a different behavior)

Reply:
Oh, I see ... we are setting the `clean` boolean now. Sorry

:man_facepalming: